src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -1,169 +1,135 @@
-    },
-    {
-      spell: SPELLS.BLACKOUT_KICK,
-      condition: describe(and(hasChi(1), not(lastSpellCast(SPELLS.BLACKOUT_KICK))), (tense) => (
+export const apl = build([
+  {
+    spell: SPELLS.TIGER_PALM,
+    condition: describe(
+      and(
+        hasResource(RESOURCE_TYPES.CHI, { atMost: 4 }),
+        hasResource(RESOURCE_TYPES.ENERGY, {
+          // over 90% of max energy
+          atLeast:
+            (BASE_ENERGY_MAX +
+              (hasTalent(TALENTS.ASCENSION_TALENT) ? ASCENSION_ENERGY_MAX_ADDITION : 0) +
+              (hasTalent(TALENTS.INNER_PEACE_TALENT) ? INNER_PEACE_ENERGY_MAX_ADDITION : 0)) *
+            0.9,
+        }),
+        not(lastSpellCast(SPELLS.TIGER_PALM)),
+      ),
+      (tense) => (
         <>
-          <SpellLink spell={SPELLS.BLACKOUT_KICK} /> was not your last cast.
+          you have at most {hasTalent(TALENTS.ASCENSION_TALENT) ? 4 : 3}{' '}
+          <SpellLink spell={RESOURCE_TYPES.CHI} />, AND about to cap energy
         </>
-      )),
-    },
-    {
-      spell: TALENTS.RUSHING_JADE_WIND_TALENT,
-      condition: hasChi(1),
-    },
-    TALENTS.CHI_WAVE_TALENT,
-    TALENTS.JADEFIRE_STOMP_TALENT,
-    {
-      spell: TALENTS.CHI_BURST_TALENT,
-      condition: and(
-        buffMissing(TALENTS.SERENITY_TALENT),
-        hasResource(RESOURCE_TYPES.CHI, { atMost: 5 }),
       ),
-    },
-  ].map((rule: Rule) => {
-    if ('condition' in rule) {
-      return { spell: rule.spell, condition: andNotSerenity(rule.condition) };
-    }
-    if ('description' in rule) {
-      return { ...rule, condition: notSerenity };
-    }
-    return { spell: rule, condition: notSerenity };
-  }),
-);
-
-export const serenityProps = (events: AnyEvent[], info: PlayerInfo): AplRuleProps => {
-  const check = aplCheck(serenityApl);
-  return {
-    apl: serenityApl,
-    checkResults: check(events, info),
-  };
-};
+    ),
+  },
+  {
+    spell: TALENTS.STRIKE_OF_THE_WINDLORD_TALENT,
+    condition: hasChi(2),
+  },
+  {
+    spell: TALENTS.WHIRLING_DRAGON_PUNCH_TALENT,
+    condition: describe(buffPresent(SPELLS.WHIRLING_DRAGON_PUNCH_USABLE), (tense) => (
+      <>
+        <SpellLink spell={TALENTS.RISING_SUN_KICK_TALENT} /> and{' '}
+        <SpellLink spell={TALENTS.FISTS_OF_FURY_TALENT} /> {tenseAlt(tense, 'are', 'were')} on
+        cooldown.
+      </>
+    )),
+  },
+  TALENTS.CELESTIAL_CONDUIT_TALENT,
+  {
+    spell: TALENTS.RISING_SUN_KICK_TALENT,
+    condition: hasChi(2),
+  },
+  {
+    spell: TALENTS.FISTS_OF_FURY_TALENT,
+    condition: hasChi(3),
+  },
+  {
+    spell: SPELLS.BLACKOUT_KICK,
+    condition: and(
+      hasChi(1),
+      buffStacks(SPELLS.TEACHINGS_OF_THE_MONASTERY, {
+        atLeast: hasTalent(TALENTS.KNOWLEDGE_OF_THE_BROKEN_TEMPLE_TALENT) ? 8 : 4,
+      }),
+    ),
+  },
+  {
+    spell: SPELLS.SPINNING_CRANE_KICK,
+    condition: buffPresent(SPELLS.DANCE_OF_CHI_JI_BUFF),
+  },
+  TALENTS.CHI_BURST_WINDWALKER_TALENT,
```

Comment:
Since chi burst is proc based now I think we need to include the condition that you have a charge